### PR TITLE
feat: start of FAQ section; integration questions

### PR DESCRIPTION
**Pull request by rightjelkin**

### CODE REVIEW

**Comment by rightjelkin** [comment]
I've also added some sc programming faqs here. Check this out

File: docs/faq/integration.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Venom Blockchain FAQ: Integration with Venom Blockchain
+
+<details>
+<summary>
+How to connect Venom Wallet to my dApp?
+</summary>
+
+The easiest way to connect your dApp to Venom Wallet is to use [Venom Connect](https://www.npmjs.com/package/venom-connect). It is a library that allows you to connect to Venom Wallet (both mobile and browser extension) and interact with it. This library provides you with a handy interface for building connect popup for our venom wallet and then gives us an interface for working with the venom network.
+
+Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/extend-our-tokensale-with-frontend.md#connecting-venom-wallet-to-your-app) paragraph of the frontend guide, that explains how to use venom-connect in your project. Moreover, you can check the final source code of this guide [here](https://github.com/venom-blockchain/guides/tree/master/tokensale-frontend).
```
Comment (adamnewton55) [inline docs/faq/integration.md:15]: The easiest way to connect your dApp to the Venom Wallet is to use [Venom Connect](https://www.npmjs.com/package/venom-connect). It is a library that allows you to connect to the Venom Wallet (both mobile and browser extension) and interact with it. This library provides you with a handy interface for building a connect popup for the venom wallet and also provides an interface for working with the venom network.

File: docs/faq/integration.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+---
+
+# Venom Blockchain FAQ: Integration with Venom Blockchain
+
+<details>
+<summary>
+How to connect Venom Wallet to my dApp?
+</summary>
+
+The easiest way to connect your dApp to Venom Wallet is to use [Venom Connect](https://www.npmjs.com/package/venom-connect). It is a library that allows you to connect to Venom Wallet (both mobile and browser extension) and interact with it. This library provides you with a handy interface for building connect popup for our venom wallet and then gives us an interface for working with the venom network.
```
Comment (adamnewton55) [inline docs/faq/integration.md:13]: The easiest way to connect your dApp to the Venom Wallet is to use [Venom Connect](https://www.npmjs.com/package/venom-connect). It is a library that allows you to connect to the Venom Wallet (both mobile and browser extension) and interact with it. This library provides you with a handy interface for building a connect popup for the venom wallet and also provides an interface for working with the venom network.

File: docs/faq/integration.md (hunk 1 above)
Comment (adamnewton55) [inline docs/faq/integration.md:15]: Check out this paragraph of the frontend guide, which explains how to use venom-connect in your project. Additionally, you can check the final source code of this guide [here](https://github.com/venom-blockchain/guides/tree/master/tokensale-frontend).

File: docs/faq/integration.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/extend-our-tokensale-with-frontend.md#connecting-venom-wallet-to-your-app) paragraph of the frontend guide, that explains how to use venom-connect in your project. Moreover, you can check the final source code of this guide [here](https://github.com/venom-blockchain/guides/tree/master/tokensale-frontend).
+
+You can read about all configuration options in venom-connect official [repository](https://github.com/web3sp/venom-connect). Also, it has an [example](https://github.com/web3sp/venom-connect/tree/main/examples/react).
```
Comment (adamnewton55) [inline docs/faq/integration.md:17]: You can read about all the configuration options in the official venom-connect [repository](https://github.com/web3sp/venom-connect). Additionally, you can view an [example](https://github.com/web3sp/venom-connect/tree/main/examples/react).

File: docs/faq/integration.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+</details>
+
+<details>
+<summary>
+What if I need very specific logic for connecting the Wallet to my dApp?
+</summary>
+
+So, in this case, you can use the library that venom-connect has been built on [inpage-provider](https://github.com/broxus/everscale-inpage-provider) and [standalone-client](https://github.com/broxus/everscale-standalone-client) - basic libraries for interaction with the venom network, so you can build your system for wallet connection. Check the documentation for these libraries in its repositories for more information.
```
Comment (adamnewton55) [inline docs/faq/integration.md:25]: In this scenario you can use the library that venom-connect has been built on [inpage-provider](https://github.com/broxus/everscale-inpage-provider) and [standalone-client](https://github.com/broxus/everscale-standalone-client) - basic libraries for interaction with the venom network. These will allow you to build your system for wallet connection. Check the documentation for these libraries in their repositories for more information.

File: docs/faq/integration.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+</details>
+
+<details>
+<summary>
+I need to call a smart contract method before the user connects the wallet. How can I do it?
+</summary>
+
+You should use [standalone-client](https://github.com/broxus/everscale-standalone-client) as a fallback for [inpage-provider](https://github.com/broxus/everscale-inpage-provider). It allows you to call smart contract's get methods without sending any transactions. Library venom-connect also gives you access to the standalone interface. You can use [getStandalone](https://github.com/web3sp/venom-connect#getstandalone) method to achieve this. You can check [this](../build/development-guides/how-to-create-your-own-non-fungible-tip-4-token/venom-in-action/frontend-for-nft-auction.md) guide when we use the standalone-client from `getStandalone` method to get the current auction information.
```
Comment (adamnewton55) [inline docs/faq/integration.md:34]: You should use the [standalone-client](https://github.com/broxus/everscale-standalone-client) as a fallback for [inpage-provider](https://github.com/broxus/everscale-inpage-provider). It allows you to call smart contract's get methods without sending any transactions. The venom-connect library also gives you access to the standalone interface. You can use the [getStandalone](https://github.com/web3sp/venom-connect#getstandalone) method to achieve this. You can check this guide, where we use the standalone-client from the getStandalone method to get the current auction information.

File: docs/faq/integration.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+You can check [this](https://github.com/broxus/ever-wallet-api) project. It will help you with transaction indexing and payment processing. It is a REST API that allows you to get information about transactions and payments. It also allows you to create payment requests and get payment notifications. You can check the documentation for this project in its repository.
+
+</details>
+
+<details>
+<summary>
+I need to have personal access to Venom's transaction history. How can I achieve this?
+</summary>
+
+You can achieve this with two modules. Both of them, in fact, is a light node of Venom Blockchain, but it has some extra interfaces for you to process incoming blocks and transaction. One of them is [ton-indexer](https://github.com/broxus/ton-indexer) and the other one is [ton-kafka-producer](https://github.com/broxus/ton-kafka-producer). The first one uses rocksdb as storage for blockchain data storage, as you can see, works with Apache Kafka.
```
Comment (adamnewton55) [inline docs/faq/integration.md:52]: You can achieve this with two modules. Both of them, in fact, are light nodes of Venom Blockchain, but have some extra interfaces for you to process incoming blocks and transactions. One of them is [ton-indexer](https://github.com/broxus/ton-indexer) and the other one is [ton-kafka-producer](https://github.com/broxus/ton-kafka-producer). The first one uses rocksdb as storage for blockchain data storage, as you can see, it works with Apache Kafka.

File: docs/faq/programming.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+Which tools can help me for developing on Venom?
+</summary>
+
+Check out [this](../build/quick-start-on-testnet.md) article. We have an awesome [repository](https://github.com/venom-blockchain/awesome-venom) as well.
+
+</details>
+
+<details>
+<summary>
+Does Venom Blockchain have some improvement proposals or some standards for example of fungible/non-fungible tokens?
```
Comment (adamnewton55) [inline docs/faq/programming.md:30]: Does Venom Blockchain have any improvement proposals or standards, for example of fungible/non-fungible tokens?

File: docs/faq/programming.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+There are many examples of this mechanic within the articles in the [Development Guides](../build/development-guides/readme.md) section. Check this out.
+
+</details>
+
+<details>
+<summary>
+What is the 'flag' of the message and which exactly should I use?
+</summary>
+
+The parameter `flag` of the message determines how much funds will be carried with the message and how to operate with the forward fee.
```
Comment (adamnewton55) [inline docs/faq/programming.md:76]: The parameter `flag` of the message determines what amount of funds will be carried with the message and how to operate with the forward fee.

File: docs/faq/programming.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+Check out [this](../build/development-guides/how-to-create-your-own-non-fungible-tip-4-token/venom-in-action/simple-nft-auction.md) guide, which shows how to implement a simple auction contract that accepts TIP-3 tokens.
+
+Of course, you need to check [TIP-3 standards](../standards/TIP-3/core-description.md) too.
+
+</details>
+
+<details>
+<summary>
+There are amount of fees including storage fees. How should I manipulate gas to be sure that my contract will be able to pay for all of them?
```
Comment (adamnewton55) [inline docs/faq/programming.md:169]: There is an amount of fees including storage fees. How should I adjust gas to be sure that my contract will be able to pay for all of them?

File: docs/faq/programming.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+</summary>
+
+You need to think out the gas management. You need to manipulate with flags of the message your contract sends and use [`tvm.rawReserve()`](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md#tvmrawreserve) function.
```
Comment (adamnewton55) [inline docs/faq/programming.md:172]: You need to plan out the gas management. You need to adjust with flags of the message your contract sends and use the [`tvm.rawReserve()`](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md#tvmrawreserve) function.

File: docs/faq/programming.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+</details>
+
+<details>
+<summary>
+There are amount of fees including storage fees. How should I manipulate gas to be sure that my contract will be able to pay for all of them?
+</summary>
+
+You need to think out the gas management. You need to manipulate with flags of the message your contract sends and use [`tvm.rawReserve()`](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md#tvmrawreserve) function. 
+
+Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/simple-tokensale.md) guide, that raises the issue of gas management.
```
Comment (adamnewton55) [inline docs/faq/programming.md:174]: Check out [this](../build/development-guides/how-to-create-your-own-fungible-tip-3-token/venom-in-action/simple-tokensale.md) guide, here we explore the issue of gas management.

File: docs/faq/readme.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+title: Venom Blockchain FAQs
+---
+
+## Venom Blockchain FAQs
+
+Use this section of Venom Documentation as a way to quickly find answers to the most popular questions about Venom Blockchain.
+
+Sections:
+
+- [Integration](./integration.md) - you can find here answers to questions about tools and ways to integrate your dApp with Venom Blockchain.
```
Comment (adamnewton55) [inline docs/faq/readme.md:13]: - [Integration](./integration.md) - here you can find answers to questions about tools and ways to integrate your dApp with Venom Blockchain.

File: docs/faq/readme.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+- [Smart Contracts Programming](./programming.md) - a collection of popular smart contracts developers question during onboarding on Venom Blockchain.
```
Comment (adamnewton55) [inline docs/faq/readme.md:14]: - [Smart Contracts Programming](./programming.md) - a collection of popular smart contracts developers questions during onboarding on Venom Blockchain.

**Pull request merged**